waltz-ng/client/data-types/routes.js

```diff
@@ -0,0 +1,42 @@
+const baseState = {
+    resolve: {
+        dataTypes: [
+            'DataTypesService',
+            (dataTypesService) => dataTypesService.loadDataTypes()
+        ]
+    }
+};
+
+
+const listState = {
+    url: 'data-types',
+    views: {'content@': require('./data-type-list') }
+};
+
+
+const viewByCodeState = {
+    url: 'data-types/code/{code}',
+    views: {'content@': require('./data-type-view') },
+    redirectTo: 'main.data-type.view',
```

Comment:
The dataTypes are passed into the injection list of the state, but it's a
case of converting the code REF_DATA to id 6000 for example.  That's what
the state gubbins in doing...

On Thu, Aug 11, 2016 at 5:39 PM, David Watkins <notifications@github.com>
wrote:

> In waltz-ng/client/data-types/routes.js
> <https://github.com/khartec/waltz/pull/417#discussion_r74457639>:
>
> > +            (dataTypesService) => dataTypesService.loadDataTypes()
> > +        ]
> > +    }
> > +};
> > +
> > +
> > +const listState = {
> > +    url: 'data-types',
> > +    views: {'content@': require('./data-type-list') }
> > +};
> > +
> > +
> > +const viewByCodeState = {
> > +    url: 'data-types/code/{code}',
> > +    views: {'content@': require('./data-type-view') },
> > +    redirectTo: 'main.data-type.view',
>
> could we not use a custom resolver for byCodeState and byIdState
> (viewState) ?
>
> e.g.
>
>
> function byCodeResolver($stateParams, dataTypeService) {
>    return {
>       dataType:   dataTypeService.findByCode($stateParams.code)
>    };
> }
>
> byCodeResovler.$inject = [...]
>
> byCodeState = {
>    ....
>    resolver: byCodeResolver
> };
>
> Then in the controllers you can receive the resolved dataTypes in the
> injection list.....
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/khartec/waltz/pull/417/files/e4bc887bdadc42f03d5e44e0876776ce30ea1a5c#r74457639>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC4529E18bAF9zDE7_ta2is0hPhJ2adoks5qe1BUgaJpZM4JiBOI>
> .
>



-- 


*Kamran Saleem*
t  +44 (0) 7732 078654
e  kamran@thinkincode.co.uk
w thinkincode.co.uk